cpp/test/sg/fil_test.cu

```diff
@@ -632,167 +633,102 @@ class TreeliteThrowSparse8FilTest : public TreeliteSparse8FilTest {
   void check() { ASSERT_THROW(setup_helper(), raft::exception); }
 };
 
-// rows, cols, nan_prob, depth, num_trees, leaf_prob, output, threshold,
-// global_bias, algo, seed, tolerance, branch comparison operator, FIL implementation, number of classes
+/** mechanism to use named aggregate initialization before C++20, and also use
+    the struct defaults. Using it directly only works if all defaulted
+    members come after ones explicitly mentioned.
+**/
+#define FIL_TEST_PARAMS(...)                  \
+  []() {                                      \
+    struct inherit_t : public FilTestParams { \
```

Comment:
It would be better to use a different name for this struct, e.g. `InitFilTestParams` or `InitializedFilTestParams`.